laceworksdk/api/content_repository.py

```diff
@@ -0,0 +1,277 @@
+# -*- coding: utf-8 -*-
+import json
+import logging
+import os
+import re
+import requests
+
+from collections import namedtuple
+from github import Github
+
+from laceworksdk.config import parse_conf_file
+
+logger = logging.getLogger(__name__)
+
+
+class ContentRepository():
+    """
+    ContentRepository API
+
+    This class loads Lacework Content Repositories to
+    facilitate "remote" acquisition of content resources.
+
+    Repositories can be loaded from a lacework.conf file or a
+    single repository can be passed during class instantiation.
+    """
+
+    REPO_STANZA_RE = re.compile('^(?:(?:content_repository)|content_repository:(.+))$')
+    DEFAULT_REPO_ID = '_default'
+    DEFAULT_REPO_TYPE = 'github'
+    DEFAULT_REPO_REF = 'main'
+
+    POLICY = namedtuple('Policy', ['id', 'uri', 'json', 'raw'])
+    QUERY = namedtuple('Query', ['id', 'uri', 'json', 'raw'])
+
+    def __init__(
+            self,
+            config_file_path=None,
+            typ=None,
+            uri=None,
+            ref=None,
+            token=None):
+        """"
+        Initializes the ContentRepository object.
+
+        Parameters:
+        config_file_path (str): File path to content repository configurations (i.e. lacework.conf)
```

Comment:
@hazedav Is this something that we should coordinate across the CLI and python SDK, or should we keep this separate? @afiune FYI ^^

The CLI stores its config at `~/.lacework.toml`

Reply:
We can totally bring in `.lacework.toml`.  I had just wired up something quick/dirty as a P.O.C.  Let me learn some more about the `.lacework.toml` and it's format and I can propose some extensions for the repository config...

Reply:
As much as I love all of this work, we need to be careful about exposing this configuration and settings to our users. The more consistent we can be, the better. 

Additionally, can we add this to the CLI already? What's stopping us from doing that instead?

I'll spend some time tomorrow reviewing this PR￼

Reply:
One approach would be to structure the configuration like this:

```
[content]
active_repository = "hazedav-private"

[[content.repositories]]
  name = "hazedav-private"
  type = "github"
  uri = "hazedav/lacework-content"
  token = "*****"
  ref = "main"
```

Which is equivelant to:
```
>>> print(json.dumps(t, indent=4))
{
    "default": {
        "account": "qa4",
        "api_key": "QA402035_B666F46CA5C749A258E59F7BD650D0F58027D81B3D166D4",
        "api_secret": "****"
    },
    "content": {
        "active_repository": "hazedav-private",
        "repositories": [
            {
                "name": "hazedav-private",
                "type": "github",
                "uri": "hazedav/lacework-content",
                "token": "****",
                "ref": "main"
            }
        ]
    }
}
```

Reply:
> As much as I love all of this work, we need to be careful about exposing this configuration and settings to our users. The more consistent we can be, the better.
> 
> Additionally, can we add this to the CLI already? What's stopping us from doing that instead?
> 
> I'll spend some time tomorrow reviewing this PR￼

I have no problem with consistency, this is my first learning of `~/.lacework.toml`, but it will be trivial to transition to this.  It seems like the python-sdk would also benefit from loading other things from this file.

With respect to "adding this to the CLI already", let's do it!  The only thing stopping us from doing that is a dedicated resource proficient in Go.  I'm working on ramping up on that now but have about 3-4 other things I'm juggling.

Regarding this PR, there is no rush here.  It's in draft status, and we can certainly revisit once we have the go-sdk/cli nailed down.

Reply:
@hazedav use me as your dedicated resource proficient in Go! I would love to help right away!! 🙌🏽

No joke, we have a Jira ticket to cover this initial discovery, lets scope some iteration to help your team on our next Sprint! (next week 🤩) https://lacework.atlassian.net/browse/ALLY-194

Reply:
Love it.  Looking forward to it.  Still gonna ramp up on Go though as I see this important to sustained collaboration...